fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/impl/networking/GlobalReceiverRegistry.java

```diff
@@ -145,17 +153,29 @@ public void endSession(AbstractNetworkAddon<H> addon) {
 		lock.lock();
 
 		try {
+			this.logTrackedAddonSize();
 			this.trackedAddons.remove(addon);
 		} finally {
 			lock.unlock();
 		}
 	}
 
+	/**
+	 * In practice, trackedAddons should never contain more than one instance.
+	 */
+	private void logTrackedAddonSize() {
+		if (LOGGER.isDebugEnabled() && this.trackedAddons.size() > 1) {
```

Comment:
Does this `size <= 1` constraint hold true on servers, given each server player has one network handler which has one network addon?

Reply:
Actually, yeah. I forgot to check about it 😅.

Reply:
I think ill drop this log to trace, to only be used for testing. I dont see a nice way to make this work for servers.